test/solvers/test_quasi_Newton.jl

```diff
@@ -357,6 +369,22 @@ using LinearAlgebra: I, eigvecs, tr, Diagonal
         @test contains(qns.direction_update.message, "gradient")
     end
 
+    @testset "Broken direction update" begin
+        M = Euclidean(2)
+        p = [0.0, 1.0]
+        f(M, p) = sum(p .^ 2)
+        grad_f(M, p) = 2 * sum(p)
+        gmp = ManifoldGradientObjective(f, grad_f)
+        mp = DefaultManoptProblem(M, gmp)
+        qns = QuasiNewtonState(
+            M, p; direction_update=QuasiNewtonGradientDirectionUpdate(ParallelTransport())
+        )
+        @test_logs (
+            :warn,
+            "Computed direction is not a descent direction; resetting to negative gradient",
+        ) match_mode = :any solve!(mp, qns)
```

Comment:
I usually leave out the string, though that might be a bit vague.

What does the `match_mode = :any` do? I have not yet seen that before.

Reply:
`match_mode = :any` checks if the string is somewhere in the warning output of the `solve!` call (IIUC without it it needs to be an exact match). How would you prefer to describe it?

Reply:
Ah if you leave the string empty it just checks that it is a warning but not the string. With containing – maybe just a key part not the full string is enough?

Reply:
I don't really have any preference here, feel free to modify this test as you see fit.

Reply:
We can also keep it as is for now.